Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -352,72 +347,78 @@ extension SwiftLanguageServer {
       return
     }
 
+    let keys = self.keys
+
     let skreq = SKRequestDictionary(sourcekitd: sourcekitd)
     skreq[keys.request] = requests.codecomplete
     skreq[keys.offset] = offset
     skreq[keys.sourcefile] = snapshot.document.url.path
     skreq[keys.sourcetext] = snapshot.text
 
-    // FIXME: SourceKit should probably cache this for us.
-    if let settings = self.buildSettingsByFile[snapshot.document.url] {
-      skreq[keys.compilerargs] = settings.compilerArguments
-    }
-
-    logAsync { _ in skreq.description }
-
-    let handle = sourcekitd.send(skreq) { [weak self] result in
+    queue.async { [weak self] in
```

Comment:
General comment: I see you're using `queue.async { [weak self]` everywhere - I don't think we need weak for these.  There is no retain cycle, since the closure itself will be released once it has been executed, which will release the reference to `self`.

Reply:
Yep, removed